src/crypto/CHIPCryptoPAL.h

```diff
@@ -308,6 +308,15 @@ CHIP_ERROR pbkdf2_sha256(const uint8_t * password, size_t plen, const uint8_t *
  **/
 CHIP_ERROR NewECPKeypair(ECPKey & pubkey, ECPKey & privkey);
 
+/** @brief Generate a new Certificate Signing Request (CSR).
+ * @param pubkey public key that'll be inserted in the CSR
+ * @param privkey private key to sign the CSR
+ * @param csr Newly generated CSR
+ * @param csr_length The caller provides the length of input buffer (csr). The function returns the actual length of generated CSR.
```

Comment:
what facility exists to find out how much space to allocate for the csr?

Reply:
Not sure if we have any meaningful way of computing it. @balducci-apple do you know any way to pre-calculate size of a CSR?

Reply:
They will be variable length (depending on the contents of the public key & signature) but with an upper bound (depending on the curve). I generated a few locally (which contained a CN of length 1) and they were 201 or 202 bytes long. Perhaps 256 bytes is more than enough to cover the basic CSR as defined here ?

Reply:
Added a constant for max CSR length. Updated the test to use that constant for buffer.

Reply:
I'd like for us to establish a pattern for such APIs.
my proposal:
for operations that can fail for reasons other than buf being to small:
```
CHIP_ERROR fillbuffer_with_variable_length_thingie(uint8_t *buf, size_t buflen, size_t &required_buflen);
```
unless something goes wrong, this form returns success even if buf is too small.  callers must check the return value and check that required_buflen <= buflen.

For functions that can't really fail, but still need to indicate how much space they needed:
```
size_t fillbuffer_with_variable_length_thingie(uint8 *buf, size_t buflen);
```
returns space required, writes to buf what it can, but does not exceed buflen.


if something goes wrong, the contents of buf and required_buflen are undefined